Sources/WalletConnectPush/Client/Wallet/ProtocolEngine/wc_pushSubscribe/PushSubscribeResponseSubscriber.swift

```diff
@@ -0,0 +1,100 @@
+
+import Foundation
+import Combine
+
+class PushSubscribeResponseSubscriber {
+    enum Errors: Error {
+        case couldNotCreateSubscription
+    }
+
+    private let networkingInteractor: NetworkInteracting
+    private let kms: KeyManagementServiceProtocol
+    private var publishers = [AnyCancellable]()
+    private let logger: ConsoleLogging
+    private let subscriptionsStore: CodableStore<PushSubscription>
+    private let groupKeychainStorage: KeychainStorageProtocol
+    private let dappsMetadataStore: CodableStore<AppMetadata>
+    private let subscriptionScopeProvider: SubscriptionScopeProvider
+    private var subscriptionPublisherSubject = PassthroughSubject<Result<PushSubscription, Error>, Never>()
+    var subscriptionPublisher: AnyPublisher<Result<PushSubscription, Error>, Never> {
+        return subscriptionPublisherSubject.eraseToAnyPublisher()
+    }
+
+    init(networkingInteractor: NetworkInteracting,
+         kms: KeyManagementServiceProtocol,
+         logger: ConsoleLogging,
+         groupKeychainStorage: KeychainStorageProtocol,
+         subscriptionsStore: CodableStore<PushSubscription>,
+         dappsMetadataStore: CodableStore<AppMetadata>,
+         subscriptionScopeProvider: SubscriptionScopeProvider
+    ) {
+        self.networkingInteractor = networkingInteractor
+        self.kms = kms
+        self.logger = logger
+        self.groupKeychainStorage = groupKeychainStorage
+        self.subscriptionsStore = subscriptionsStore
+        self.dappsMetadataStore = dappsMetadataStore
+        self.subscriptionScopeProvider = subscriptionScopeProvider
+        subscribeForSubscriptionResponse()
+    }
+
+    private func subscribeForSubscriptionResponse() {
+        let protocolMethod = PushSubscribeProtocolMethod()
+        networkingInteractor.responseSubscription(on: protocolMethod)
+            .sink {[unowned self] (payload: ResponseSubscriptionPayload<SubscriptionJWTPayload.Wrapper, SubscribeResponseParams>) in
+                Task(priority: .high) {
+                    logger.debug("Received Push Subscribe response")
+
+                    guard let responseKeys = kms.getAgreementSecret(for: payload.topic) else {
+                        logger.debug("PushSubscribeResponseSubscriber: no symmetric key for topic \(payload.topic)")
+                        subscriptionPublisherSubject.send(.failure(Errors.couldNotCreateSubscription))
+                        return
+                    }
+
+                    // get keypair Y
+                    let pubKeyY = responseKeys.publicKey
+                    let peerPubKeyZ = payload.response.publicKey
+
+                    var account: Account!
+                    var metadata: AppMetadata!
+                    var pushSubscriptionTopic: String!
+                    var availableScope: Set<NotificationType>!
+                    let (_, claims) = try SubscriptionJWTPayload.decodeAndVerify(from: payload.request)
```

Comment:
you can decode SubscriptionJWTPayload in first tuple variable. Does it help here?

Reply:
not sure, what you mean...